packages/flutter/test/cupertino/action_sheet_test.dart

```diff
@@ -83,6 +87,88 @@ void main() {
     await gesture.up();
   });
 
+  testWidgets('Button appearance is correct with text scaling', (WidgetTester tester) async {
+    // Verifies layout of action button in various text scaling by drawing it in
+    // all 12 iOS text scale in one golden image.
+
+    // The following function returns a CupertinoActionSheetAction that:
+    // * Has a fixed width
+    // * Is unconstrained in height
+    // * Is aligned center in a grid of fixed height
+    // * Is surrounded by a black border
+    const double buttonWidth = 400;
+    const double rowHeight = 100;
+    Widget testButton(double contextBodySize) {
+      const double standardHigBody = 17.0;
+      final double contextScaleFactor = contextBodySize / standardHigBody;
+      return OverrideMediaQuery(
+        transformer: (MediaQueryData data) {
+          return data.copyWith(
+            textScaler: TextScaler.linear(contextScaleFactor),
+          );
+        },
+        child: SizedBox(
+          height: rowHeight,
+          child: Center(
+            child: UnconstrainedBox(
+              child: ConstrainedBox(
+                constraints: const BoxConstraints.tightFor(width: buttonWidth),
+                child: DecoratedBox(
+                  decoration: BoxDecoration(border: Border.all()),
+                  child: CupertinoActionSheetAction(
+                    onPressed: () {},
+                    child: const Text('Button'),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(CupertinoApp(
+      home: CupertinoPageScaffold(
+          child: Center(
+            child: Column(
+              children: <Widget>[
+                Row(children: <Widget>[
+                  /*xs*/ testButton(14),
+                  /*s*/  testButton(15),
+                ]),
+                Row(children: <Widget>[
+                  /*m*/  testButton(16),
+                  /*l*/  testButton(17),
+                ]),
+                Row(children: <Widget>[
+                  /*xl*/ testButton(19),
+                  /*xxl*/testButton(21),
+                ]),
+                Row(children: <Widget>[
+                  /*xxxl*/testButton(23),
+                  /*ax1*/ testButton(28),
+                ]),
+                Row(children: <Widget>[
+                  /*ax2*/ testButton(33),
+                  /*ax3*/ testButton(40),
+                ]),
+                Row(children: <Widget>[
+                  /*ax4*/ testButton(47),
+                  /*ax5*/ testButton(53),
+                ]),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await expectLater(
+      find.byType(Column),
+      matchesGoldenFile('cupertinoActionSheet.textScaling.png'),
```

Comment:
Apart from the golden test, I'd consider also trying to add an explicit expectation about the font sizes. That'd help make sure that the data you gathered are taken into account in future changes, and that we don't accidentally break fidelity because the visual difference was small.

For example:
```suggestion
    final Iterable<RichText> buttons = tester.widgetList<RichText>(
      find.text('Button', findRichText: true));
    final Iterable<double?> sizes = buttons.map((RichText text) {
      return text.textScaler.scale(text.text.style!.fontSize!);
    });
    expect(sizes,
      <double>[21, 21, 21, 21, 23, 24, 24, 28, 33, 40, 47, 53].map(
        (double size) => moreOrLessEquals(size, epsilon: 0.001)),
    );

    await expectLater(
      find.byType(Column),
      matchesGoldenFile('cupertinoActionSheet.textScaling.png'),
```

Reply:
… And in fact, when I run that test on the current PR revision, it actually finds a discrepancy!

```
Expected: MappedListIterable<double, Matcher>:[
            <21.0 (±0.001)>,
            <21.0 (±0.001)>,
            <21.0 (±0.001)>,
            <21.0 (±0.001)>,
            <23.0 (±0.001)>,
            <24.0 (±0.001)>,
            <24.0 (±0.001)>,
            <28.0 (±0.001)>,
            <33.0 (±0.001)>,
            <40.0 (±0.001)>,
            <47.0 (±0.001)>,
            <53.0 (±0.001)>
          ]
  Actual: MappedIterable<RichText, double?>:[
            21.0,
            21.0,
            21.0,
            21.0,
            24.0,
            24.0,
            25.0,
            28.0,
            33.0,
            40.0,
            47.0,
            53.0
          ]
   Which: at location [4] is <24.0> which 24.0 is not in the range of 23.0 (±0.001).
```

Two discrepancies, at sizes xl and xxxl.

So I think that test expectation definitely pulls its weight.

Reply:
Thank you so much! I wanted to do this as well but struggled to come up with a way.